crates/cli/src/main.rs

```diff
@@ -113,8 +113,33 @@ fn main() -> io::Result<()> {
                 false => CodeGenBackend::Llvm(LlvmBackendMode::BinaryGlue),
             };
 
+            let link_type = LinkType::Dylib;
+            let linking_strategy =
+                if matches.get_one::<String>(FLAG_LINKER).map(|s| s.as_str()) == Some("legacy") {
+                    LinkingStrategy::Legacy
+                } else {
+                    let target = Triple::host().into();
+                    match roc_linker::support_level(link_type, target) {
+                        roc_linker::SupportLevel::Full => LinkingStrategy::Surgical,
+                        roc_linker::SupportLevel::Wip => {
+                            println!(
+                                "Warning! Using an unfinished surgical linker for target {target}"
+                            );
+                            LinkingStrategy::Surgical
+                        }
+                        roc_linker::SupportLevel::None => LinkingStrategy::Legacy,
+                    }
+                };
```

Comment:
Can share a function with the code in `lib.rs`? I think they are the same or nearly the same.